artemis-journal/src/main/java/org/apache/activemq/artemis/core/journal/impl/JournalImpl.java

```diff
@@ -1858,6 +1988,8 @@ public synchronized void compact() throws Exception {
          journalLock.writeLock().unlock();
       }
 
+      processBackup();
```

Comment:
I thought during our discussions we want the backup before compaction so its literally as raw as it gets.

Reply:
Or am i miss understanding whats going on here?

Reply:
processBackup() will copy the BKP files to the retention folder.

Before I do the computation I need any BKP files cleared so they can be treated. So, the backup is before compacting, hence I'm calling processBackup before compacting here.

Reply:
Understood